{"schemaVersion":{"patch":0,"minor":3,"major":0},"abstract":[{"type":"text","text":"Try to cancel the execution of the repeated task."}],"metadata":{"externalID":"s:7NIOCore12RepeatedTaskC6cancel7promiseyAA16EventLoopPromiseVyytGSg_tF","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"cancel"},{"kind":"text","text":"("},{"kind":"externalParam","text":"promise"},{"kind":"text","text":": "},{"kind":"typeIdentifier","preciseIdentifier":"s:7NIOCore16EventLoopPromiseV","text":"EventLoopPromise"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","preciseIdentifier":"s:s4Voida","text":"Void"},{"kind":"text","text":">?)"}],"modules":[{"name":"NIOCore"}],"symbolKind":"method","title":"cancel(promise:)","roleHeading":"Instance Method","role":"symbol"},"kind":"symbol","hierarchy":{"paths":[["doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore","doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/RepeatedTask"]]},"variants":[{"paths":["\/documentation\/niocore\/repeatedtask\/cancel(promise:)"],"traits":[{"interfaceLanguage":"swift"}]}],"identifier":{"interfaceLanguage":"swift","url":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/RepeatedTask\/cancel(promise:)"},"sections":[],"primaryContentSections":[{"kind":"declarations","declarations":[{"languages":["swift"],"platforms":["Linux"],"tokens":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"cancel"},{"kind":"text","text":"("},{"kind":"externalParam","text":"promise"},{"kind":"text","text":": "},{"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/EventLoopPromise","kind":"typeIdentifier","preciseIdentifier":"s:7NIOCore16EventLoopPromiseV","text":"EventLoopPromise"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","preciseIdentifier":"s:s4Voida","text":"Void"},{"kind":"text","text":">? = nil)"}]}]},{"content":[{"anchor":"discussion","level":2,"type":"heading","text":"Discussion"},{"inlineContent":[{"text":"Whether the execution of the task is immediately canceled depends on whether the execution of a task has already begun.","type":"text"},{"text":" ","type":"text"},{"text":"This means immediate cancellation is not guaranteed.","type":"text"}],"type":"paragraph"},{"inlineContent":[{"text":"The safest way to cancel is by using the passed reference of ","type":"text"},{"code":"RepeatedTask","type":"codeVoice"},{"text":" inside the task closure.","type":"text"}],"type":"paragraph"},{"inlineContent":[{"type":"text","text":"If the promise parameter is not "},{"type":"codeVoice","code":"nil"},{"type":"text","text":", the passed promise is fulfilled when cancellation is complete."},{"type":"text","text":" "},{"type":"text","text":"Passing a promise does not prevent fulfillment of any promise provided on original task creation."}],"type":"paragraph"}],"kind":"content"}],"references":{"doc://co.pointfree.VaporRouting/documentation/NIOCore/RepeatedTask/cancel(promise:)":{"url":"\/documentation\/niocore\/repeatedtask\/cancel(promise:)","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"cancel"},{"kind":"text","text":"("},{"kind":"externalParam","text":"promise"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"EventLoopPromise","preciseIdentifier":"s:7NIOCore16EventLoopPromiseV"},{"kind":"text","text":"<"},{"kind":"typeIdentifier","text":"Void","preciseIdentifier":"s:s4Voida"},{"kind":"text","text":">?)"}],"kind":"symbol","title":"cancel(promise:)","identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/RepeatedTask\/cancel(promise:)","role":"symbol","type":"topic","abstract":[{"type":"text","text":"Try to cancel the execution of the repeated task."}]},"doc://co.pointfree.VaporRouting/documentation/NIOCore":{"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore","role":"collection","kind":"symbol","type":"topic","abstract":[],"title":"NIOCore","url":"\/documentation\/niocore"},"doc://co.pointfree.VaporRouting/documentation/NIOCore/RepeatedTask":{"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/RepeatedTask","abstract":[{"text":"Returned once a task was scheduled to be repeatedly executed on the ","type":"text"},{"code":"EventLoop","type":"codeVoice"},{"text":".","type":"text"}],"title":"RepeatedTask","kind":"symbol","role":"symbol","type":"topic","url":"\/documentation\/niocore\/repeatedtask","fragments":[{"text":"class","kind":"keyword"},{"text":" ","kind":"text"},{"text":"RepeatedTask","kind":"identifier"}],"navigatorTitle":[{"text":"RepeatedTask","kind":"identifier"}]},"doc://co.pointfree.VaporRouting/documentation/NIOCore/EventLoopPromise":{"fragments":[{"kind":"keyword","text":"struct"},{"kind":"text","text":" "},{"kind":"identifier","text":"EventLoopPromise"}],"abstract":[{"type":"text","text":"A promise to provide a result later."}],"navigatorTitle":[{"kind":"identifier","text":"EventLoopPromise"}],"identifier":"doc:\/\/co.pointfree.VaporRouting\/documentation\/NIOCore\/EventLoopPromise","title":"EventLoopPromise","role":"symbol","type":"topic","url":"\/documentation\/niocore\/eventlooppromise","kind":"symbol"}}}